components/ruby/lib/chef-licensing/restful_client/cache_manager.rb

```diff
@@ -0,0 +1,48 @@
+require "active_support/cache"
+require "tmpdir" unless defined?(Dir.mktmpdir)
+require_relative "../config"
+
+module ChefLicensing
+  module RestfulClient
+    class CacheManager
+      DEFAULT_TTL = 60
+      def initialize(cache_dir = nil)
+        @cache = ActiveSupport::Cache::FileStore.new(cache_dir || Dir.tmpdir)
```

Comment:
I don't see the need to provide an optional `cache_dir`. Do you have any particular use case? if not I would prefer abstracting it out.

Reply:
I am using the `cache_dir` during testing.

Reply:
you can stub it